contracts/Option.sol

```diff
@@ -147,55 +140,6 @@ contract Option is IOption, IERC20, European, Ownable {
         _burn(msg.sender, options);
     }
 
-    /**
-     * @notice Exercises an option after `expiryTime` but before `expiryTime + windowSize`.
-     * Requires a transaction inclusion proof which is verified by our chain relay.
-     * @dev Can only be called by the parent factory contract.
-     * @param seller Account to exercise against
-     * @param height Bitcoin block height
-     * @param index Bitcoin tx index
-     * @param txid Bitcoin transaction id
-     * @param proof Bitcoin inclusion proof
-     * @param rawtx Bitcoin raw tx
-     **/
-    function executeExercise(
```

Comment:
Why did you move the executeExercise out of the option contract?

Reply:
The core logic only pertains to the obligation contract so there is no need to do the forward call in the option contract. For verification we needed `btcHash` and `format` anyway which added additional back and forth complexity.

Reply:
As dicussed, the user should have one contract to call when exercising => either the option or the obligation contract.